Pokemon Unity/Assets/Scripts2/Pokemon/Pokemon.cs

```diff
@@ -269,6 +265,70 @@ public Pokemon(Pokemons pokemon) : this()
 
             //calcStats();
         }
+
+        public Pokemon(string nickName, int form,
+            Pokemons species, Abilities ability,
+            Nature nature,
+            bool isShiny, bool? gender,
+            int[] pokerus, int pokerusStrain,
+            int? shadowLevel,
+            int currentHp, Items item,
+            byte[] iv, byte[] ev, 
+            int obtainedLevel, int currentLevel, int currentExp,
+            int happiness, Status status, int statusCount,
+            int eggSteps, Items ballUsed,
+            Item.Item.Mail mail, Move[] moves,
```

Comment:
mail string, not mail class. Item id holds the item id used in the mail class... can always use `if mail string is not null, and item id is mail letter => new Mail(item, string)`

Reply:
I think `if item.category == mail` might work better, because regardless if `mail.message` string is null or empty, it would still need to be created as a `new Mail()`...